docs/dictionary.rst

```diff
@@ -0,0 +1,1101 @@
+
+Further reading: `<https://www.docker.com/resources/what-container/>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Docker image:
+
+Docker image
+------------
+	A read-only file that represents a filesystem that contains some sort of code and that code's dependencies. A Docker image can be created using the `docker build` command in conjunction with a :ref:`dict Dockerfile`. If a workflow language references a Docker image, then the workflow executor will download that Docker image (unless was already downloaded previously) and add a writable layer onto the Docker image, which results in the creation of a :ref:`dict Docker container`.  
+
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Dockerfile:
+
+Dockerfile
+----------
+	A file describing the creation of a :ref:`dict Docker image` by running commands that each form a :ref:`dict layer`.  
+
+Further reading: `<https://docs.docker.com/engine/reference/builder/>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Dockstore CLI:
+
+Dockstore CLI
+-------------
+*abbreviation for* Dockstore Command Line Interface  
+
+	A command-line program developed by Dockstore. It is not required to use Dockstore, but it has many features to make running and developing workflows easier.  
+
+see also :ref:`dict CLI`  
+
+Further reading: `<https://docs.dockstore.org/en/stable/advanced-topics/dockstore-cli/dockstore-cli-faq.html>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Dockstore GitHub App:
+
+Dockstore GitHub App
+--------------------
+	The GitHub App that allows for Dockstore to communicate with GitHub repositories.  
+
+see also :ref:`dict GitHub App registration`  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict DOI:
+
+DOI
+---
+*abbreviation for* Digital Object Identifier  
+
+	An identifier that provides a long-lasting link to some sort of :ref:`dict immutable` digital object. On Dockstore, you can use Zenodo to mint a DOI of your workflows and tools to increase reproducibility.  
+
+
+.. updated 2022-05-16  
+
+
+
+.. _dict DS-I Africa:
+
+DS-I Africa
+-----------
+*abbreviation for* Data Science for health discovery and Innovation in Africa  
+
+	An :ref:`dict NIH` initiative to leverage data science to address the African continent's public health needs.  
+
+Further reading: `<https://commonfund.nih.gov/africadata>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict EC2:
+
+EC2
+---
+*abbreviation for* Elastic Compute Cloud  
+
+	The cloud computing side of :ref:`dict AWS`. When running workflows on these backends, disk size will scale with your workflow requirements automatically. EC2 instances allow you to make use of Amazon's :ref:`dict spot instance` feature, which may reduce the cost of running workflows.  
```

Comment:
1. I don't know about disk sizes scaling automatically. I don't think it's an EC2 feature, but I could be wrong.
2. EC2 instances can be spot instances, I wouldn't say they allow you to make use of them.

Reply:
EC2 can autoscale and Amazon markets this feature heavily, but I might be slightly incorrect on the disk size. This is what [Cromwell says about disk size on AWS](https://cromwell.readthedocs.io/en/stable/RuntimeAttributes/):

>For the AWS Batch backend, the disk volume is managed by AWS EBS with autoscaling capabilities. As such, the Disk _[sic]_ size and disk type will be ignored. If provided, the mount point will be verified at runtime.

I was conflating EBS with EC2, although strictly speaking EBS works alongside EC2. I don't know for sure if AWS Batch is the only way to run workflows on EC2... how would you word this?


Regarding spot instances, if you choose an AWS backend for your Seven Bridges workspace, you can select spot instances. I've used them myself, and there's docs here: https://sb-biodatacatalyst.readme.io/docs/about-spot-instances
Looks like TOIL supports it too (search [this page](https://toil.readthedocs.io/en/3.15.0/running/cloud/amazon.html) for "spot instance")
But unlike [preemptibles](https://cromwell.readthedocs.io/en/stable/RuntimeAttributes/#preemptible), this isn't something you put into a runtime attribute in CWL/WDL.

Reply:
I've decided to nix the bit about automatically scaling, since I'm not 100% sure on that.